libs/dataset-explorer/src/lib/DefaultTab.tsx

```diff
@@ -0,0 +1,389 @@
+              </div>
+              {canRenderChart ? (
+                <AccessibleChart plotlyProps={plotlyProps} theme={getTheme()} />
+              ) : (
+                <MissingParametersPlaceholder>
+                  {localization.Interpret.ValidationErrors.datasizeError}
+                </MissingParametersPlaceholder>
+              )}
+            </div>
+            <div className={classNames.horizontalAxisWithPadding}>
+              <div className={classNames.paddingDiv} />
+              <div className={classNames.horizontalAxis}>
+                <div>
+                  <DefaultButton
+                    onClick={this.setXOpen.bind(this, true)}
+                    text={
+                      this.context.jointDataset.metaDict[
+                        this.state.chartProps.xAxis.property
+                      ].abbridgedLabel
+                    }
+                    title={
+                      this.context.jointDataset.metaDict[
+                        this.state.chartProps.xAxis.property
+                      ].label
+                    }
+                  />
+                </div>
+              </div>
+            </div>
+          </div>
+          <SidePanel
+            chartProps={this.state.chartProps}
+            cohorts={this.context.errorCohorts.map(
+              (errorCohort) => errorCohort.cohort
+            )}
+            jointDataset={this.context.jointDataset}
+            selectedCohortIndex={this.state.selectedCohortIndex}
+            setColorOpen={this.setColorOpen}
+            onChartTypeChange={this.onChartTypeChange}
+          />
+        </div>
+      </div>
+    );
+  }
+
+  private setSelectedCohort = (
+    _: React.FormEvent<HTMLDivElement>,
+    item?: IDropdownOption
+  ): void => {
+    if (item?.key !== undefined) {
+      this.setState({ selectedCohortIndex: item.key as number });
+    }
+  };
+
+  private onChartTypeChange = (
+    _ev?: React.SyntheticEvent<HTMLElement>,
+    item?: IChoiceGroupOption
+  ): void => {
+    const newProps = _.cloneDeep(this.state.chartProps);
+    if (item?.key === undefined || !newProps) {
+      return;
+    }
+    newProps.chartType = item.key as ChartTypes;
+    if (newProps.yAxis.property === ColumnCategories.None) {
+      newProps.yAxis = this.generateDefaultYAxis();
+    }
+    this.setState({ chartProps: newProps });
+  };
+
+  private readonly setXOpen = (val: boolean): void => {
+    if (val && this.state.xDialogOpen === false) {
+      this.setState({ xDialogOpen: true });
+      return;
+    }
+    this.setState({ xDialogOpen: false });
+  };
+
+  private readonly setColorOpen = (): void => {
+    this.setState({ colorDialogOpen: true });
+  };
+
+  private readonly setColorClose = (): void => {
+    this.setState({ colorDialogOpen: false });
+  };
+
+  private readonly setYOpen = (val: boolean): void => {
+    if (val && this.state.yDialogOpen === false) {
+      this.setState({ yDialogOpen: true });
+      return;
```

Comment:
Maybe just use if/else instead of doing an early return?  I think it's a bit easier to follow the code that way.
Even better, you can assign a variable like this:
```
const yDialogOpen = val && this.state.yDialogOpen === false
this.setState({ yDialogOpen: yDialogOpen });
```

That reduces 5 lines to 2 lines, and I think it is easier to follow.